### ISSUE TITLE
False positive for importing a metaclass

### ISSUE THREAD

**Issue by marmistrz** [PyCQA/pylint#1630]
### Steps to reproduce
Lint the following code:
```
from abc import ABCMeta

class foo(meta=ABCMeta):
    pass
```

### Current behavior
```
W:  1, 0: Unused ABCMeta imported from abc (unused-import)
```

### Expected behavior
No `unused-import` error.

### pylint --version output
```
pylint 1.7.2, 
astroid 1.5.3
Python 3.6.2 (default, Jul 20 2017, 03:52:27) 
[GCC 7.1.1 20170630]
```

**Comment by rogalski** [comment]
It's kind of corner case, we successfully lint code below, which is correct declaration of using metaclass: 

```python
from abc import ABCMeta

class foo(metaclass=ABCMeta):
    pass

```

**Comment by Wesitos** [comment]
Still present in version 1.8.2
I wouldn't consider this a corner case considering that python's 3.6 new `__init_subclass__` method supports arbitrary keyword argument passing

**Comment by jonapich** [comment]
I'm also facing this, in an `__init_subclass__` scenario. (pylint 2.2.2)

**Comment by Kane610** [comment]
This example will get 

> subclass.py:1:0: W0611: Unused TEXT imported from baseclass (unused-import)

file baseclass.py
```
TEXT = "string"
class baseclass:
    def __init_subclass__(cls, text, **kwargs):
        super().__init_subclass__(**kwargs)

```
file subclass.py

```
from baseclass import TEXT, baseclass
class subclass(baseclass, text=TEXT):
    pass
```

**Comment by douglas-raillard-arm** [comment]
Here is another variation on the same theme without using `__init_subclass__`, which also triggers:
```
test.py:2:0: W0611: Unused import os (unused-import)
```

```
import os

class Meta(type):
    def __new__(cls, name, bases, dct, *args, x=None, **kwargs):
        return super().__new__(cls, name, bases, dct, *args, **kwargs)

class C(metaclass=Meta, x=os):
    pass
```

**Comment by hippo91** [comment]
@douglas-raillard-arm thanks for this update.

**Comment by rsiemens** [comment]
I'm also seeing this when using an imported module in an expression in the class arguments.

```
$ python -m pylint --version
pylint 2.13.9
astroid 2.11.5
Python 3.9.7 (default, Oct 22 2021, 07:42:57)
[Clang 12.0.5 (clang-1205.0.22.11)]
```

```
import django

class SomeMetaClass(type):
    pass

class MyClass(metaclass=type if django.VERSION >= (3, 2) else SomeMetaClass):
    pass
```

```
$ python -m pylint bug.py
************* Module bug
...
bug.py:1:0: W0611: Unused import django (unused-import)
```

**Comment by jacobtylerwalls** [comment]
Only the last example in https://github.com/PyCQA/pylint/issues/1630#issuecomment-1126424923 seems to still reproduce on 2.14 dev, so I'm going to refocus the issue title.

**Comment by samskiter** [comment]
@jacobtylerwalls 
Also seems to repro for an inner class definition :

```
from abc import ABCMeta
class Piece:
    class Connector(metaclass=ABCMeta):
        pass
```
